evm/src/cpu/kernel/functions/storage_read.asm

```diff
@@ -0,0 +1,12 @@
+// TODO: Dummy code for now.
+function storage_read {
```

Comment:
I don’t think that this `function` syntax is a good fit here for a few reasons.

Firstly, it’s duplicative. In most asm languages, the start of a function would be denoted with a simple label, i.e.:
```
storage_read:
    JUMPDEST
    PUSH 1234
    ...
```
This is enough to make a function call.

Secondly, it actually reduces flexibility. Suppose you want to be able to call into the _middle_ of a function (there’s many reasons why you’d want to do this! you may want to begin execution in the middle of a loop or have two functions that differ only in how they begin, so you want them to share code). This syntax does not permit you to do this.

Finally, the `}` would implicitly insert a `RETURN` in high-level programming language, but it doesn’t here. It _cannot_ implicity insert it here because not all functions end in a return! The last bytes of a function might handle a conditional case that then jumps backward, or the function might make a tail call to another function. So this notation is a bit misleading.

Reply:
I suspect the reason why you’ve chosen this syntax to distinguish between global labels and local labels.

In most asm, local labels are not local to the _function_ but local to the assembly file, so this is an unusual thing to do.

I’m sure there’s another way to distinguish global and local labels. E.g. you can force the programmer to write `g.foo` for a global and `l.bar` for a local.

Note that having per-file local labels means that each file has to be assembled separately. It is no longer correct to concatenate them all and assemble the one long file.

Reply:
Changed it as we discussed, using the `global <label>:` syntax